vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -899,6 +901,10 @@ public List<String> getWeekdays() {
          * @return this instance for method chaining
          */
         public DatePickerI18n setWeekdays(List<String> weekdays) {
+            if (weekdays != null && weekdays.size() != 7) {
+                LoggerFactory.getLogger(getClass().getName()).warn(String.format("setWeekdays input list should have exactly 7 elements. Instead got %s", weekdays.size()));
```

Comment:
- `getClass` is enough to get the logger
- "parameter list" might be more appropriate to reference the method parameter than "input list"
- Not sure how you bypassed the formatting check in the CI, it should have complained 😅. Anyway, before committing your code, try to remember to run `mvn formatter:format`.